avocado_vt/vt_agent/src/avocado_vt/agent/core/rpc/server.py

```
@@ -0,0 +1,251 @@
+
+    def __init__(self, addr, **kwargs):
+        """Initialize the custom XML-RPC server with enhanced error handling."""
+        try:
+            super().__init__(addr, **kwargs)
+            self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        except OSError as e:
+            if "Address already in use" in str(e):
+                raise OSError(
+                    f"Cannot bind to {addr[0]}:{addr[1]} - address already in use. "
+                    "Another agent instance may be running."
+                ) from e
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)